card/mod/05_standard/set/type/cardtype.rb

```diff
@@ -11,7 +11,22 @@
       %{<div>Sorry, this card must remain a Cardtype so long as there are <strong>#{ card.name }</strong> cards.</div>}
     else
       super args
-    end  
+    end
+  end
+
+  view :add_button do |args|
+    args[:title] ||= "Add #{card.name}"
+    if args[:params]
+      context = ((@parent && @parent.card) || card).name
+      args[:params].gsub!(/=([^&]+)/) do |match|
+        "=#{$1.to_name.to_absolute context}"
+      end
+    end
+    %{
+      <a class='btn btn-default' href='/new/#{card.key}?#{args[:params]}'>
+        #{ _render_title args }
+      </a>
+    }
   end
```

Comment:
I like it!  (Are we still thinking to add add_link view, too? (and, I guess, add_url, add_path...)